client/requestUtil.js

```diff
@@ -81,10 +81,20 @@ RequestUtil.prototype.refreshAWSCredentials = function () {
   }
   return window.fetch(url, params)
     .then((response) => {
-      if (!response.ok) {
+      if (response.ok) {
+        return response.arrayBuffer()
+      }
+      if (response.status === 400) {
+        // Bad request
+        return response.text().then((text) => {
+          if (text === 'Signed request body of the client timestamp is required.') {
```

Comment:
Why can't we just throw new Error(\`Credential server response ${response.status}: ${text}\`) when not ok? Response text can be changed from time to time

Reply:
I thought we cannot trust the data from the network :) . But with your question, I understand this is our server, so we can. Thanks.

Reply:
@darkdh force-pushed the updated code